File: admin-backend/admin-system/src/main/java/io/github/faustofan/admin/system/domain/model/SysUser.java
package io.github.faustofan.admin.system.domain.model;

import io.github.faustofan.admin.shared.persistence.model.TenantAware;
import io.github.faustofan.admin.system.domain.enums.UserStatus;
import org.babyfish.jimmer.sql.*;
import org.jspecify.annotations.Nullable;

import java.util.List;
import java.util.Map;

/**
 * 系统用户实体
 */
@Entity
@Table(name = "sys_user")
public interface SysUser extends TenantAware {

    @Key
    String username();

    String password(); // 记得加密存储

    @Nullable
    String nickname();

    @Nullable
    String email();

    @Nullable
    String mobile();

    // 归属部门
    @ManyToOne
    @JoinColumn(name = "org_id")
    SysOrg org();

    @Column(name = "is_super_admin")
    boolean isSuperAdmin();

    // ABAC 核心：用户自身的动态属性 (Postgres JSONB -> Java Map)
    @Serialized
    @Nullable
    Map<String, Object> attributes();

    // 多对多关联角色
    @ManyToMany
    @JoinTable(
            name = "sys_user_role",
            joinColumnName = "user_id",
            inverseJoinColumnName = "role_id"
    )
    List<SysRole> roles();

    // 修改为枚举类型
    @Column(name = "status")
    @Default("ACTIVE")
    UserStatus status();

}

File: src/main/kotlin/io/github/faustofan/admin/system/domain/entity/SysUser.kt
package io.github.faustofan.admin.system.domain.entity

import io.github.faustofan.admin.common.model.TenantAware
import org.babyfish.jimmer.sql.*

@Entity
@Table(name = "sys_user")
interface SysUser : TenantAware {

    @Key
    val username: String

    val nickname: String?

    // 密码 (加密存储)
    val passwordHash: String

    val email: String?

    val phone: String?

    val avatar: String?

    // 状态
    val status: UserStatus

    // 关联：所属部门 (多对一)
    @ManyToOne
    val org: SysOrg?

    // 关联：拥有的角色 (多对多)
    @ManyToMany
    @JoinTable(
        name = "sys_user_role_mapping",
        joinColumnName = "user_id",
        inverseJoinColumnName = "role_id"
    )
    val roles: List<SysRole>
}

/**
 * 用户状态枚举
 */
@EnumType(EnumType.Strategy.NAME)
enum class UserStatus {
    ACTIVE,
    LOCKED,
    DISABLED
}
